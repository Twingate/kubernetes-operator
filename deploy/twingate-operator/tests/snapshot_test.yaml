suite: Snapshot
excludeTemplates:
  - NOTES.txt # NOTES.txt is empty
tests:
  - it: should render
    values:
      - values.test.yaml
    asserts:
      - matchSnapshot: {}
