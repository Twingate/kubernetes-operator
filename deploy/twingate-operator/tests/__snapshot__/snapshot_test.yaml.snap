should render:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-twingate-operator-role-cluster
    rules:
      - apiGroups:
          - apiextensions.k8s.io
        resources:
          - customresourcedefinitions
        verbs:
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - list
          - watch
      - apiGroups:
          - ""
          - events.k8s.io
        resources:
          - events
        verbs:
          - '*'
      - apiGroups:
          - admissionregistration.k8s.io/v1
          - admissionregistration.k8s.io/v1beta1
        resources:
          - validatingwebhookconfigurations
          - mutatingwebhookconfigurations
        verbs:
          - create
          - patch
      - apiGroups:
          - twingate.com
        resources:
          - '*'
        verbs:
          - '*'
      - apiGroups:
          - '*'
        resources:
          - pods
          - services
          - secrets
          - services/status
        verbs:
          - list
          - watch
          - patch
          - get
          - create
          - delete
      - apiGroups:
          - apps
        resources:
          - deployments
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-twingate-operator-rolebinding-cluster
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-twingate-operator-role-cluster
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-twingate-operator
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
      name: RELEASE-NAME-twingate-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: twingate-operator
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: twingate-operator
        spec:
          containers:
            - command:
                - kopf
                - run
                - ./main.py
                - -A
                - --standalone
                - --liveness=http://0.0.0.0:8080/healthz
                - --log-format=full
              env:
                - name: TWINGATE_API_KEY
                  valueFrom:
                    secretKeyRef:
                      key: TWINGATE_API_KEY
                      name: RELEASE-NAME-twingate-operator
                - name: TWINGATE_NETWORK
                  value: <network slug>
                - name: TWINGATE_HOST
                  value: twingate.com
                - name: TWINGATE_DEFAULT_RESOURCE_TAGS
                  value: '{}'
                - name: TWINGATE_REMOTE_NETWORK_ID
                  value: <remote network id>
              image: twingate/kubernetes-operator:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8080
              name: twingate-operator
              readinessProbe:
                httpGet:
                  path: /healthz
                  port: 8080
              resources: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-twingate-operator
  4: |
    apiVersion: v1
    data:
      TWINGATE_API_KEY: PGFwaSBrZXk+
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
      name: RELEASE-NAME-twingate-operator
      namespace: NAMESPACE
  5: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
      name: RELEASE-NAME-twingate-operator
      namespace: NAMESPACE
